tests/e2e/utils/data_fetcher/logs.py

```diff
@@ -0,0 +1,39 @@
+import json
+from functools import lru_cache
+from typing import List, Optional, Union
+
+from mypy_boto3_cloudwatch import CloudWatchClient
+from pydantic import BaseModel
+from retry import retry
+
+
+class Log(BaseModel):
+    level: str
+    location: str
+    message: Union[dict, str]
+    timestamp: str
+    service: str
+    cold_start: Optional[bool]
+    function_name: Optional[str]
+    function_memory_size: Optional[str]
+    function_arn: Optional[str]
+    function_request_id: Optional[str]
+    xray_trace_id: Optional[str]
+    extra_info: Optional[str]
+
+
+@lru_cache(maxsize=10, typed=False)
+@retry(ValueError, delay=1, jitter=1, tries=20)
```

Comment:
I think we may adjust this number similar way you did here: https://github.com/awslabs/aws-lambda-powertools-python/pull/1457/files#diff-ae1df3bf5a60a4705238094f7903173aa2111914c9571300fb2c97d80a6bd150R12